### PR TITLE
destructure props, used fragments and conditional rendering

### PR DESCRIPTION
**Pull request by riteshv-webelight**
did Conditional Rendering using && operator, Ternary operator and if..else
Destructuring Props 
React Fragments
Conditionally setting classname and text

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -73,15 +73,28 @@ function Header() {
 
 // Menu component
 function Menu() {
```
Comment (webclues-manish) [inline src/index.js]: create arrow function

File: src/index.js [hunk 2]
```diff
@@ -73,15 +73,28 @@ function Header() {
...
+  const pizzas = pizzaData;
```
Comment (webclues-manish) [inline src/index.js]: no need to create new const

File: src/index.js [hunk 3]
```diff
@@ -94,14 +107,17 @@ function Menu() {
 }
 
 // single pizza list item component
-function Pizza(props) {
+function Pizza({ pizzaObj }) {
```
Comment (webclues-manish) [inline src/index.js]: create arrow function

File: src/index.js [hunk 4]
```diff
@@ -94,14 +107,17 @@ function Menu() {
...
+  // Conditional rendering using multiple returns, eg.if..else
+  // if (pizzaObj.soldOut) return null;
+
   return (
-    <li className="pizza">
-      <img src={props.pizzaObj.photoName} alt={props.pizzaObj.name} />
+    <li className={`pizza ${pizzaObj.soldOut ? "sold-out" : ""}`}>
```
Comment (webclues-manish) [inline src/index.js]: destructure pizzaObj above and then use the properies/keys here

File: src/index.js
Comment (webclues-manish) [inline src/index.js:1]: create separate files for APP and all other components

Comment (riteshv-webelight) [reply src/index.js:1]: Created separate js file for header, footer and menu components and imported them in index.js file

File: src/index.js (hunk 1 above)
Comment (riteshv-webelight) [reply src/index.js]: Created Arrow Function for all components

File: src/index.js (hunk 2 above)
Comment (riteshv-webelight) [reply src/index.js]: Removed unnecessary const and used pizzaData object directly

File: src/index.js (hunk 4 above)
Comment (riteshv-webelight) [reply src/index.js]: Yep, that was very useful. destructured all the values and used them directly

File: src/footer.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+// Footer components
+const Footer = () => {
+  const hour = new Date().getHours();
+  const openingHour = 10;
+  const closingHour = 22;
+  const isOpen = hour >= openingHour && hour <= closingHour;
+  console.log(isOpen);
```
Comment (webclues-manish) [inline src/footer.js]: remove console

File: src/footer.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+// Footer components
+const Footer = () => {
```
Comment (webclues-manish) [inline src/footer.js]:
```suggestion
export const Footer = () => {
```

File: src/footer.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+          {" "}
+          We're happy to welcome you between {openingHour}:00 and {closingHour}
+          :00.
+        </p>
+      )}
+    </footer>
+  );
+};
+
+export default Footer;
```
Comment (webclues-manish) [inline src/footer.js]: remove

File: src/header.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+// Header Component
+const Header = () => {
```
Comment (webclues-manish) [inline src/header.js]:
```suggestion
export const Header = () => {
```

File: src/header.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+  return (
+    <header className="header">
+      <h1>Fast React Pizza Co.</h1>
+    </header>
+  );
+};
+
+export default Header;
```
Comment (webclues-manish) [inline src/header.js]: remove

File: src/index.js [hunk 10]
```diff
@@ -51,76 +55,16 @@ const pizzaData = [
...
+const App = () => {
   // const c = console.log("");
   return (
     <div className="container">
       <Header />
-      <Menu />
+      <Menu pizzaData={pizzaData} />
       <Footer />
     </div>
   );
```
Comment (webclues-manish) [inline src/index.js]: create a separate file for App component too, and we shouild not export default components

File: src/menu.js
Comment (webclues-manish) [inline src/menu.js:1]: remove unneccessary commented code

File: src/menu.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+      {pizzaData.length > 0 && (
+        // {/* React Fragment - when we don't want to wrap two elements in one div */}
+        <>
+          <p>
+            Authentic italian cuisin. 6 creative dishes to choose from. All from
+            our stone own, all organic, all delicious.
+          </p>
+          <ul className="pizzas">
+            {pizzaData.map((pizza) => (
+              <Pizza pizzaObj={pizza} key={pizza.name} />
```
Comment (webclues-manish) [inline src/menu.js]: why you are passing key prop from here while you are already passing whole pizza objec to Pizza component

File: src/menu.js [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+        pizzaName={pizzaData[0].name}
+        ingredients={pizzaData[0].ingredients}
+        price={pizzaData[0].price}
+      /> */}
+    </main>
+  );
+};
+
+// single pizza list item component
+const Pizza = ({ pizzaObj }) => {
```
Comment (webclues-manish) [inline src/menu.js]: although I asked to remove the key prop, but if you had added there and not extracted it here, why ?

key don't work like that.

always create separate file for new comonents

File: src/menu.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+      <div>
+        <h3>{name}</h3>
+        <p>{ingredients}</p>
+        <span>{soldOut ? "SOLD OUT" : price}</span>
+      </div>
+    </li>
+  );
+};
+
+export default Menu;
```
Comment (webclues-manish) [inline src/menu.js]: we should not export default

File: src/pizzaList.js [hunk 14]
```diff
@@ -0,0 +1,15 @@
+// single pizza list item component
+export const Pizza = ({ pizzaObj }) => {
+  const { name, ingredients, price, photoName, soldOut } = pizzaObj;
+
+  return (
+    <li className={`pizza ${soldOut ? "sold-out" : ""}`}>
```
Comment (webclues-manish) [inline src/pizzaList.js]: add a key attribute here

File: src/pizzaList.js [hunk 15]
```diff
@@ -0,0 +1,15 @@
+// single pizza list item component
+export const Pizza = ({ pizzaObj }) => {
+  const { name, ingredients, price, photoName, soldOut } = pizzaObj;
+
+  return (
+    <li className={`pizza ${soldOut ? "sold-out" : ""}`} key={pizzaObj.name}>
```
Comment (webclues-manish) [inline src/pizzaList.js]:
```suggestion
    <li className={`pizza ${soldOut ? "sold-out" : ""}`} key={name}>
```

**Review by webclues-manish — APPROVED** [review]

**Pull request merged**